src/DSharpPlus.Examples.Commands.Basics/Program.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using DSharpPlus.Commands;
+using DSharpPlus.Commands.Processors.SlashCommands;
+using DSharpPlus.Commands.Processors.TextCommands;
+using DSharpPlus.Commands.Processors.TextCommands.Parsing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DSharpPlus.Examples.Commands.Basics
+{
+    public static class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            ConfigurationBuilder configurationBuilder = new();
+            configurationBuilder.AddEnvironmentVariables("");
+            configurationBuilder.AddCommandLine(args);
+
+            IConfiguration configuration = configurationBuilder.Build();
+            ServiceCollection serviceCollection = new();
+            serviceCollection.AddSingleton(configuration);
+            Assembly currentAssembly = typeof(Program).Assembly;
+
+            serviceCollection.AddSingleton(async serviceProvider =>
+            {
+                DiscordClient client = new(new DiscordConfiguration()
+                {
+                    Token = configuration.GetValue<string>("discord:token") ??
+                            throw new InvalidOperationException("Missing Discord token."),
+                    Intents = DiscordIntents.AllUnprivileged | DiscordIntents.MessageContents | TextCommandProcessor.RequiredIntents
+                });
+
+                CommandsExtension extension = client.UseCommands(new()
+                {
+                    DebugGuildId = configuration.GetValue<ulong?>("discord:debug_guild_id", null),
+                    ServiceProvider = serviceProvider,
+                    
+                });
+                await extension.AddProcessorAsync(new TextCommandProcessor());
+                await extension.AddProcessorAsync(new SlashCommandProcessor());
+                extension.AddCommands(currentAssembly);
+
+                return client;
+            });
+            IServiceProvider serviceProvider = serviceCollection.BuildServiceProvider();
+
+            DiscordClient client = await serviceProvider.GetRequiredService<Task<DiscordClient>>();
```

Comment:
why is this async

Reply:
asked by lunar in last review

Reply:
@OoLunar why

Reply:
https://github.com/DSharpPlus/Example-Bots/pull/13#discussion_r1474888211
i think this comment is meant which was to a different part of the code

Reply:
No, this is intended. Adding processors is an asynchronous operation. Encouraging `.GetAwaiter().GetResult()` goes against everything I've worked on within the lib, especially when it's done on `ValueTask`s.

Reply:
I would do the whole setup outside the `AddSinglton` and then add the client